File: 프로그래머스/lv0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.py
def solution(lines):
    li = ['0']*200
    for i in range(len(lines)):
        for j in range(i + 1, len(lines)):
            s = 100 + max(lines[i][0], lines[j][0])
            e = 100 + min(lines[i][1], lines[j][1])
            if e > s:
                li[s:e] = ['1'] * (e - s)
    return "".join(li).count('1')

File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> lines) {
    int answer = 0;
    vector<int> cnt(200, 0);
    for (vector<int>& line: lines) {
        for (int n=line[0]; n<line[1]; n++) {
            cnt[n+100]++;
        }
    }
    for (int& n: cnt) {
        if (n > 1) answer++;
    }
    return answer;
}